files/bootstrap.sh

```diff
@@ -195,8 +195,7 @@ if [ -z "$CLUSTER_NAME" ]; then
     exit  1
 fi
 
-ZONE=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
-AWS_DEFAULT_REGION=$(echo $ZONE | awk '{print substr($0, 1, length($0)-1)}')
+AWS_DEFAULT_REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq .region -r)
```

Comment:
To make @nckturner's life a bit easier, can we use IMDSv2 for this? Something like this:
```bash
TOKEN=$(curl -X PUT -H "X-aws-ec2-metadata-token-ttl-seconds: 600" "http://169.254.169.254/latest/api/token")
IMDS_CURL_ARGS="-s -H \"X-aws-ec2-metadata-token: $TOKEN\""
AWS_DEFAULT_REGION=$(curl $IMDS_CURL_ARGS http://169.254.169.254/latest/dynamic/instance-identity/document | jq .region -r)
AWS_SERVICES_DOMAIN=$(curl $IMDS_CURL_ARGS http://169.254.169.254/2018-09-24/meta-data/services/domain)
```

Also, can you confirm that `jq` is always available?

Reply:
Haha, for sure. I will upgrade to IMDSv2 directly in this PR to save time on potential migration. 

We install jq here and have some references in bootstrap.sh.
https://github.com/awslabs/amazon-eks-ami/blob/06cb77298984d84fc0edd0a876fb571b576d6ae8/scripts/install-worker.sh#L61